### PR TITLE
General cleanup, flake8 monster PR

### PR DESCRIPTION
**Pull request by dannon**
This is a work in progress towards the goal of making the entire codebase pass flake8 so that we have a clean slate to validate new code against moving forward.  We'd love for people to go ahead and start reviewing and testing.

If you want to contribute and clean up an additional module or script simply remove it from the blacklist in flake8 sources and run .ci/flake8_wrapper.sh to see what the errors are.

Much of the code here lacks test coverage, so any eyes we can get on it would be great.

### CODE REVIEW

**Comment by dannon** [comment]
-1, please don't merge yet.

I guess it's required that I add justification here, so: 

At best, it changes nothing for the end user to rush and release it in 15.07.  This is a purely developer side improvement (and a really nice one, at that).  At worst, we get bugs in the release.

We've already caught several major bugs since it was merged last time (admittedly mostly the same kind of bugs, of two particular classes -- missing imports and improper use of sqlalchemy filters).  This is a _massive_ PR, and it hasn't had the eyes it needs on it, yet -- the history multipanel didn't work at all until additional commits of mine this morning, for example.  Much of the content here isn't actually covered under testing (which is another issue, though not one we'll address with the PR), so that's not a reliable method for validation.

The only cost to delaying this post-freeze is that it's possible we might have a few slightly annoying merges to deal with when applying fixes to #release_15.07 and merging them forward to -dev, where this will have been applied.  I volunteer to do all of these, if someone has a change and doesn't want to deal with it.

TLDR:  Buys us nothing to rush it into the freeze, except bugs.  I'd feel much more comfortable giving this a full dev cycle of tire-kicking.  We didn't even want to bump an egg just prior to release, last time around, and this touches _way_ more functionality.

**Comment by natefoo** [comment]
FWIW, I am still +1 for including this in 15.07.

**Comment by jmchilton** [comment]
Also +1 for including it in 15.07.

I appreciate justification given by @dannon - I don't agree with the calculus but I respect it and it is technically valid. I am however concerned that the "possible presence of bugs" results in a -1 and not a -0. This is an incredibly low bar for a veto that should be in my opinion used very sparingly given its absurd power. This sets a bad precedent for the future - any commit could potentially have bugs.

**Comment by dannon** [comment]
I totally understand what you're saying regarding 'possible existence of bugs'.  If the freeze wasn't today, I would have a different opinion here I think.  That said, I'd probably qualify this as 'likely existence of bugs', given what we've already fixed since the last merge.

It should probably matter as well that one of the pull request authors and someone who has attempted to thoroughly validate and fix this is not comfortable.

Lastly, I don't want to speak for them and this doesn't count for anything technically, but I believe in the various threads and irc convos @martenson and @carlfeberhard were either -1 or -0.

Edit:  from irc, should xpost it here
@jmchilton I wholeheartedly agree with avoiding the capricious veto, but I feel like I've been backed into a corner a little bit here with how the last merge happened
(Not that anyone has done anything wrong at all, but given the way the rules are currently stated I didn't know another way to specify WIP and prevent merge.  @natefoo  suggested the self -1)

**Comment by natefoo** [comment]
A bit of context, the -1 thing was indeed from earlier today on IRC. I was pointing out that if reopened, it was likely that someone would un-[WIP], +1 and merge, since the only issue with the previous PR was that it was marked [WIP] and given an implicit +1 merge, so if @dannon wanted to prevent this from being merged, he'd have to -1 it.

**Comment by bgruening** [comment]
I spend most of my time in Dublin and today in testing this and so far this looks good to me. The last glitch with multiple-history was already fixed (my fix didn't made it in - why do airplanes do not have internet connection ;)), thanks @dannon!

That said, either we commit it now'ish and start testing on a few real life instances. I would be willing to make this switch here in Freiburg. Or we branch Galaxy as soon as possible and commit this to dev right after branching.
@natefoo when is the branching to next-stable scheduled? Will the new wheel stuff be included?

I'm a little bit worried about the `revert the revert part` as far as I can see @remimarenco is not listed as author but he was also part of this adventure.

**Comment by natefoo** [comment]
@bgruening The release_15.07 branch will be created today, I was working through PRs yesterday and today to get as many merged as was reasonable.

The wheels PR will not be included, there is still a lot of work to do on it.

**Comment by nsoranzo** [comment]
@bgruening The original commits have been already been merged, so they are in the history. Unfortunately the first revert seems to make a rebase over it impossible, maybe cherry-picking could work, but I guess it's too late now.

**Comment by nsoranzo** [comment]
I've added other 2 commits, can go out of WIP for me. Needs to be updated to merge cleanly.

I'm going on vacation for 2 weeks, I'll leave the beast to you, guys :)

**Comment by carlfeberhard** [comment]
@dannon agreed I am against having this in 15.07.
@jmchilton I see what you're saying about precedent, but at 500 files and +6,133 −5,476 lines - this isn't just any pull request.

Don't get me wrong: I think the changes needed to happen and are awesome. Thanks to all who worked on it! @nsoranzo @remimarenco especially.

I also still don't understand: why does it need to be in the release?

**Comment by jmchilton** [comment]
@carlfeberhard The release branch has been branched - this will not be in 15.07.

@dannon Any chance you can remove your -1 now?

**Comment by dannon** [comment]
-1 retracted.  There's still more to clean up, but it wouldn't hurt to go ahead and get this into -dev to work against.

**Comment by nsoranzo** [comment]
:+1:

**Comment by bgruening** [comment]
:+1:

**Comment by bgruening** [comment]
Wuhu! Thanks so much to you all for making this happen!
A lot of work went into this PR and it was a great community effort! Thanks!!!
